src/lib/sw.js

```diff
@@ -117,29 +93,21 @@ workboxRouting.registerRoute(
 // Configure default cache for standard web.dev files: the offline page, various images etc.
 precacheAndRoute(manifest);
 
-/**
- * Match "/foo-bar/ "and "/foo-bar/as/many/of-these-as-you-like/" (with optional trailing
- * "index.html"), normal page nodes for web.dev. This only matches on pathname.
- */
-const contentPathRe = new RegExp("^/([\\w-]+/)*(|index.html)$");
-
 /**
  * Match "/foo-bar" and "/foo-bar/as-many/but/no/trailing/slash" (but not "/foo/bar/index.html").
  * This only matches on pathname (so it must always start with "/").
  */
 const untrailedContentPathRe = new RegExp("^(/[\\w-]+)+$");
 
 /**
- * Send normal nodes to cache first.
+ * Match fetches for patials, for SPA requests. Matches "/foo-bar/index.json" and
+ * "/foo-bar/many/parts/index.json", for partial SPA requests.
  */
-workboxRouting.registerRoute(
-  ({url, event}) => {
-    return url.host === self.location.host && contentPathRe.test(url.pathname);
-  },
-  new workboxStrategies.NetworkFirst({
-    plugins: [normalizeIndexCacheKeyPlugin],
-  }),
-);
+const partialPathRe = new RegExp("^/([\\w-]+/)*index\\.json$");
+const partialStrategy = new workboxStrategies.NetworkFirst();
```

Comment:
I'd recommend adding in a `cacheName` here that used something like semver semantics, like

```
const partialStrategy = new workboxStrategies.NetworkFirst({
  cacheName: 'partial-cache-v1',
  plugins: [/* see comment about expiration */],
});
```

If you at some point in the future make changes to your surrounding HTML that end up being incompatible with the currently cached JSON content, you'd bump that version.

This would make it easy to purge all of the out-of-date JSON as part of an `activate` handler, and you wouldn't have to worry about mismatches in the stitching-together process due to stale data.

Reply:
Also sent as part of #2270. I presume this will invalidate our users' previous caches which have an implicit name. If you have advice on that I'd appreciate it over in #2270.

Reply:
And by invalidate, I mean unref, so they'll still be around.